recipes/libnghttp3/all/conanfile.py

```diff
@@ -0,0 +1,88 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.apple import is_apple_os
+from conan.tools.cmake import CMake, CMakeToolchain, CMakeDeps, cmake_layout
+from conan.tools.files import get, rmdir, copy
+from conan.tools.gnu import PkgConfigDeps
+from conan.tools.microsoft import is_msvc
+from conan.tools.scm import Version
+import os
+
+
+required_conan_version = ">=1.53.0"
+
+
+class Nghttp3Conan(ConanFile):
+    name = "libnghttp3"
+    description = "HTTP/2 C Library and tools"
+    topics = ("http", "http3")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://nghttp2.org/nghttp3/"
+    license = "MIT"
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+
+    def build_requirements(self):
+        self.tool_requires("cmake/[>=3.20 <4]")
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def validate(self):
+        if self.settings.compiler == "gcc" and Version(self.settings.compiler.version) < "6":
```

Comment:
please, in case this project needs C++14, use the template for validation:

https://github.com/conan-io/conan-center-index/blob/master/docs/package_templates/cmake_package/all/conanfile.py#L78

Reply:
This is a C project. It shouldn't have any build dependency on C++14.

Reply:
I guess I should say the qpack examples require C++-17; however, this recipe explicitly disables the examples.

Reply:
So why does it need GCC 6? Does it use C-11 ?

Reply:
Since their build files were ported from ngtcp2 (which were ported from nghttp2), I assumed this needed to stay the same as nhgttp2. I can remove it and see if it breaks.